src/specimens/Html.js

```diff
@@ -77,27 +120,50 @@ class Html extends React.Component {
   constructor() {
     super();
     this.state = {
-      viewSource: false
+      viewSource: false,
+      parentWidth: 0,
+      screenSize: SCREENSIZES[0]
     };
   }
 
   componentDidMount() {
-    const {runScript} = this.props;
+    const {runScript, frame, device, catalog: {theme}} = this.props;
+    let devices = theme.devices ? getScreensizes( theme.devices ) : SCREENSIZES;
     if (runScript) {
       Array.from(this.refs.specimen.querySelectorAll('script'))
         .forEach(runscript);
+    } else if (frame || device) {
+      this.getWidth();
+      device !== 'select' ? this.setSize( devices.find(sc=>sc.name === device) || SCREENSIZES[0] ) : null;
     }
   }
 
+  getWidth() {
+    this.setState({parentWidth: this.refs.specimen.getBoundingClientRect().width});
+  }
+
+  toggleSource() {
+    this.setState({viewSource: !this.state.viewSource});
```

Comment:
When updating the state based on the current one, do it with the callback which makes sure you get the current state `this.setState(({foo}) => ({foo: !foo}))`